restrict rule snippets to single turn user inputs (#6607)
RasaHQ/rasa: docs/docs/policies.mdx

training data. See the [Rules documentation](./rules.mdx) for further information
on how to define rules.

> You can remove rule restriction and check for contradictions by setting the following

Reviewer: Can this be explained in two separate sub-sections? That way there can be a full definition of what each param does (e.g. what is the rule restriction?) and why you would use it.

training data. See the [Rules documentation](./rules.mdx) for further information
on how to define rules.

> The `RulePolicy` has the following configuration options:

Reviewer 2: ```suggestion
The RulePolicy has the following configuration options:
```

```yaml title="config.yml"
policies:
  - name: "RulePolicy"
    core_fallback_threshold: 0.3
    core_fallback_action_name: action_default_fallback
    enable_fallback_prediction: true
    restrict_rules: true
    check_for_contradictions: true
```

* `core_fallback_threshold` (default: `0.3`): Please see the
   [fallback documentation](fallback-handoff.mdx#handling-low-core-confidence) for
   further information.
* `core_fallback_action_name` (default: `action_default_fallback`): Please see the
   [fallback documentation](fallback-handoff.mdx#handling-low-core-confidence) for
   further information.
* `enable_fallback_prediction` (default: `true`): Please see the
   [fallback documentation](fallback-handoff.mdx#handling-low-core-confidence) for
   further information.

> * `check_for_contradictions` (default: `true`): After training, the `RulePolicy` will

Reviewer 2: ```suggestion
* `check_for_contradictions` (default: `true`): After training, the RulePolicy will
```

>    perform a check to make sure that the rules don't contradict the stories and each

Reviewer 2: ```suggestion
   perform a check to make sure that there are no rules that contradict with each other, or with any stories.
```

>    other. The following snippet is an example for two contradicting rules:

Reviewer 2: ```suggestion
   The following snippet is an example for two contradicting rules:
```

    ```yaml-rasa
    rules:
    - rule: Chitchat
      steps:
      - intent: chitchat
      - action: utter_chitchat

    - rule: Greet instead of chitchat
      steps:
      - intent: chitchat
      - action: utter_greet  # `utter_greet` contradicts `utter_chitchat` from the rule above
    ```
 * `restrict_rules` (default: `true`): Rules are restricted to one user turn, but
    there can be multiple bot events, including e.g. a form being filled and its

>     subsequent submission. It is **not recommended** to change this parameter to

Reviewer 2: ```suggestion
    Change this parameter to `false` at your own risk.
```

>     `false`. Overusing rules for purposes out of the [recommended use cases](rules.mdx)

Reviewer 2: ```suggestion
    Overusing rules for purposes outside of the [recommended use cases](rules.mdx)
```

The `Rule Policy` is a policy that handles conversation parts that follow
a fixed behavior. Please see [Rules](./rules.mdx) for further information.

> You can remove rule restriction and check for contradictions by setting the following

Reviewer 3: do we actually want to advertise this? (cc @akelad )

If we add this, we need to first explain what the "restrictions" are.

Reviewer 2: we should document it, yes. But +1 to explaining what the restrictions are

Author: the restriction is explained in `rules.mdx`

Reviewer 3: then let's link there
